transaction-service-api/src/main/java/pl/adamboguszewski/transactionservice/service/api/transaction/CreateTransactionRequest.java

```diff
@@ -0,0 +1,55 @@
+package pl.adamboguszewski.transactionservice.service.api.transaction;
+
+import lombok.Value;
+import lombok.NonNull;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+@Value
+public class CreateTransactionRequest {
+
+    @NonNull
+    Long totalPrice;
+
+    @NonNull
+    List<Product> products;
+
+    @NonNull
+    Info transactionInfo;
+
+    @Value
+    public static class Info {
+
+        @NonNull
+        LocalDateTime transactionTime;
+
+        @NonNull
+        UUID checkoutId;
+
+        @NonNull
+        List<PaymentInfo> paymentsInfo;
+
+        @Value
+        public static class PaymentInfo {
```

Comment:
You cut TransactionInfo to Info (line 23) and then cut TransactionPayment to PaymentInfo.